sources/api/thar-be-updates/src/status.rs

```diff
@@ -0,0 +1,391 @@
+use crate::error;
+use crate::error::Result;
+use bottlerocket_release::BottlerocketRelease;
+use chrono::{DateTime, Utc};
+use model::modeled_types::FriendlyVersion;
+use serde::{Deserialize, Serialize};
+use signpost::State;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::convert::TryInto;
+use std::fs::File;
+use std::os::unix::process::ExitStatusExt;
+use std::process::Output;
+
+pub const UPDATE_STATUS_FILE: &str = "/run/cache/thar-be-updates/status.json";
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum UpdateState {
+    Idle,
+    Available,
+    Staged,
+    Ready,
+}
+
+/// UpdateImage represents a Bottlerocket update image
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct UpdateImage {
+    arch: String,
+    version: semver::Version,
+    variant: String,
+}
+
+impl UpdateImage {
+    pub fn version(&self) -> &semver::Version {
+        &self.version
+    }
+}
+
+/// StagedImage represents a Bottlerocket image that is written to a partition set
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct StagedImage {
+    image: UpdateImage,
+    // Indicates whether this image is marked for next boot
+    next_to_boot: bool,
+}
+
+impl StagedImage {
+    pub(crate) fn set_next_to_boot(&mut self, next_to_boot: bool) {
+        self.next_to_boot = next_to_boot
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum CommandStatus {
+    Success,
+    Failed,
+    Unknown,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum UpdateCommand {
+    RefreshUpdate,
+    PrepareUpdate,
+    ActivateUpdate,
+    DeactivateUpdate,
+}
+
+/// CommandResult represents the result of an issued command
+#[derive(Debug, Clone, Deserialize, Serialize)]
+struct CommandResult {
+    cmd_type: UpdateCommand,
+    cmd_status: CommandStatus,
+    timestamp: DateTime<Utc>,
+    exit_status: Option<i32>,
+    stderr: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct UpdateStatus<T> {
+    update_state: UpdateState,
+    available_updates: Vec<T>,
```

Comment:
I do need the information like `variant`, `arch`, `max_version` for when I update the staging partition information or check for the latest update. `available_updates` stores the update information as `Vec<update_metadata::Update>` for that purpose.

Maybe instead of just the version, `available_updates` should also store the `arch` and `variant`, `max_version`?

Reply:
Can we store what we need but not display it?  Or store it in a separate struct that's used only where we need it?

Reply:
Ah, I missed the `simplify_update_info` call where we're doing just that.